doc/techsupport-manager/core_file_manager.md

```diff
@@ -0,0 +1,329 @@
+* [Unit Test](#unit-test)
+
+# List of Tables
+
+[Table 1: Abbreviations](#table-1-abbreviations)
+
+# Revision
+
+Rev   |   Date   |  Author   | Change Description
+:---: | :-----:  | :------:  | :---------
+1.0   | 05/07/19 | Kalimuthu | Initial version
+2.0   | 03/08/19 | Rajendra  | Review Comments 
+
+
+# About this Manual
+
+This document describes a core file management framework in SONiC network operating system. This feature helps to manage the  generation of core files  and exporting them to an external server for better debuggability.
+
+# Scope
+
+This document describes the following enhancements to the SONiC Core File handling: -
+
+ - The Debian "systemd-coredump" service is added to the SONiC host O/S, replacing the existing SONiC core dump handling. This service can be managed through the associated "coredumpctl" tool and "coredump.conf".
+ - A SONiC service called core file manager is added to provide additional management and export of the core dumps that systemd-coredump generates to a remote host.
+ - The core file manager service also includes "show tech-support" data along with the core file when exporting crash reports to remote host. Exporting of crash reports is done both periodically and on discovery of a new core file.
+ - SONiC UI capabilities are added to manage the core file manager service and systemd-coredump functionality.
+This document describes the high level design details of core file manager framework that consists of core file and tech-support data generation framework. It also provides the exporting of  tech-support data to a remote server when a new core is generated and also on a periodic interval.  This document describes the tools used to support the core file generation and export functionality.
+
+# Definition/Abbreviation
+
+### Table 1: Abbreviations
+
+
+| **Term**     |  **Meaning**                  |
+|:-------------|:------------------------------|
+| tech-support | Technical support data        |
+| SCP          | Secure Copy Protocol          |
+| SFTP         | Secure File Transfer Protocol |
+
+
+## Requirements Overview
+
+### Functional Requirements
+
+This document describes new mechanisms to manage the core files that are generated in a SONiC environment. When a process is terminated unexpectedly, the System generates the core dump of that process in most cases. The core files are stored on the persistent storage like SSDs. These core files need to be managed on the SONiC devices to optimize disk space usage, exported to remote server for offline analysis and recorded to maintain a history of occurrences. In addition, existing "tech-support" information is associated with the core dumps and managed similarly.
+
+### Core Manager Requirements
+
+  1. Add a new SONiC service to manage the core dumps as follows: - 
+
+	 a.  Support per-process core file rotation and archiving to optimize disk space 
+
+ 	 b.  Strip Core files sensitive information 
```

Comment:
Which module does this? Debian service or ...?
Can you please share some details on, how it works?
Is there a way to enable/disable it?

Reply:
New service named 'coredumpctl.service' is added to monitor the per process core file rotation and archive managment. Through systemctl command, this can be enabled/disabled.

Reply:
Can this service support user to enable core on few processes instead all at once ? Basically can we collect core per process?